mgmtworker/cloudify_system_workflows/search_utils.py

```diff
@@ -123,6 +126,34 @@ def get_node_instances(self, node_instance, **kwargs):
                                                _get_all_results=True,
                                                constraints=kwargs)
 
+    def get_operation_names(self, operation_name, **kwargs):
+        try:
+            deployment_id = kwargs.pop('deployment_id')
+        except KeyError:
+            raise NonRecoverableError(
+                "You should provide 'deployment_id' when getting operation "
+                "names.  Make sure you have `deployment_id` constraint "
+                "declared for your 'operation_name' parameter.")
```

Comment:
This can be more concise, something like `Parameters of type 'operation_name' require the 'deployment_name' constraint`

Also I wonder if we can put something useful in this error message? The operation name? The path would've been more useful, but I don't think we have that here...